hrpc/query.go

```diff
@@ -97,6 +98,15 @@ func (bq *baseQuery) setCacheBlocks(cacheBlocks bool) {
 func (bq *baseQuery) setConsistency(consistency ConsistencyType) {
 	bq.consistency = consistency
 }
+func (bq *baseQuery) setPriority(priority uint32) {
+	bq.priority = priority
+}
+func (bq *baseQuery) Priority() *uint32 {
```

Comment:
Returning by point here is weird. I understand that when we marshal to proto it wants pointers-to-int and maybe doing it this way avoids an allocation (because otherwise in `marshalProto()` you'd have to create an `int` and take its address thereby causing a heap allocation of a new integer), but from an API perspective it's odd. Did you do it this way to avoid an allocation?

Reply:
Yeah, to avoid an allocation.

Reply:
And now this has become slightly awkward in #268, which fixes something I missed here.

Reply:
Oh I didn't notice that. I think returning just uint32 make more sense. Returning a pointer or a uint32 is the same in the end

Reply:
It's same except for the added allocation. The extra allocation is not a big deal, so I'll just clean this up in #268.

Reply:
Updated #268